ckanext/data_qld/test_plugin.py

```diff
@@ -0,0 +1,62 @@
+# encoding: utf-8
+import os
+
+import ckantoolkit as tk
+from werkzeug.datastructures import FileStorage
+
+from ckan import plugins
+
+from ckanext.data_qld.tests.conftest import DatasetFactory, ResourceFactory
+
+
+class DataQldTestPlugin(plugins.SingletonPlugin):
+    plugins.implements(plugins.IActions)
+
+    # IActions
+
+    def get_actions(self):
+        return {
+            'qld_test_create_dataset': qld_test_create_dataset,
+            'qld_test_purge_dataset': qld_test_purge_dataset,
+            'qld_test_patch_dataset': qld_test_patch_dataset,
+            'qld_test_create_resource_for_dataset': qld_test_create_resource_for_dataset
+        }
+
+
+@tk.side_effect_free
```

Comment:
How is a call that creates a dataset supposed to be "side effect free"?

Reply:
Leve it for now.